search/api.py

```diff
@@ -0,0 +1,207 @@
+        index=settings.ELASTICSEARCH_INDEX,
+        doc_type=PROFILE_DOC_TYPE,
+    )
+
+    if len(errors) > 0:
+        raise ReindexException("Error during bulk insert: {errors}".format(
+            errors=errors
+        ))
+    refresh_index()
+
+    return insert_count
+
+
+def index_profiles(profiles, chunk_size=100):
+    """
+    Add/update profile records in Elasticsearch.
+
+    Args:
+        profiles (iterable of Profile):
+            Iterable of profiles
+        chunk_size (int):
+            How many profiles to index at once.
+
+    Returns:
+        int: Number of indexed items
+    """
+    # Use an iterator so we can keep track of what's been indexed already
+    profiles = iter(profiles)
+
+    count = 0
+    chunk = list(islice(profiles, chunk_size))
+    while len(chunk) > 0:
+        count += _index_profiles_chunk(chunk)
+        chunk = list(islice(profiles, chunk_size))
+
+    refresh_index()
+
+    return count
+
+
+def remove_profile(profile):
+    """
+    Remove a profile from Elasticsearch.
+    """
+    conn = get_conn()
+    try:
+        conn.delete(index=settings.ELASTICSEARCH_INDEX, doc_type=PROFILE_DOC_TYPE, id=profile.id)
+    except NotFoundError:
+        # Item is already gone
+        pass
+
+
+def serialize_profile(profile):
+    """
+    Serializes profile for use with Elasticsearch.
+
+    Args:
+        profile (Profile): A profile to serialize
+    Returns:
+        dict: The data to be sent to Elasticsearch
+    """
+    serialized = ProfileSerializer().to_representation(profile)
+    serialized['id'] = profile.id
+    serialized['_id'] = profile.id  # For Elasticsearch
+    return serialized
+
+
+def refresh_index():
+    """
+    Refresh the Elasticsearch index
+    """
+    get_conn().indices.refresh(index=settings.ELASTICSEARCH_INDEX)
+
+
+def create_profile_mapping():
+    """
+    Create a mapping for profiles. If one already exists, delete it first.
+    """
+    conn = get_conn(verify=False)
+
+    index_name = settings.ELASTICSEARCH_INDEX
+    if conn.indices.exists_type(index=index_name, doc_type=PROFILE_DOC_TYPE):
+        conn.indices.delete_mapping(index=index_name, doc_type=PROFILE_DOC_TYPE)
+
+    mapping = Mapping(PROFILE_DOC_TYPE)
+    mapping.field("username", "string", index="not_analyzed")
+    mapping.field("education", "nested", properties={
+        'degree_name': {'type': 'string', 'index': 'not_analyzed'}
+    })
+    mapping.save(index_name)
+
+
+def create_mappings():
+    """
+    Create all mappings, deleting existing mappings if they exist.
+    """
+    create_profile_mapping()
+
+
+def clear_index():
+    """
+    Wipe and recreate index and mapping. No indexing is done.
+    """
+    conn = get_conn(verify=False)
+    index_name = settings.ELASTICSEARCH_INDEX
+    if conn.indices.exists(index_name):
+        conn.indices.delete(index_name)
+    conn.indices.create(index_name)
+    conn.indices.refresh()
+
```

Comment:
sorry to harp on the empty lines across these files, but this one and line 206 don't seem to add any value